### PR TITLE
feat: Disable slash commands for encrypted rooms

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

Currently, several apps and bots don’t work properly in E2EE rooms
So, it is needed to make clear to users that it is not possible to use apps and slash commands if E2EE is enabled in the room.
## Further comments

[E2EE-24](https://rocketchat.atlassian.net/browse/E2EE2-24)

[E2EE-24]: https://rocketchat.atlassian.net/browse/E2EE-24?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
don't some slash commands work just fine?  Like those that generate google meet etc.  The messages generated of course aren't encrypted.. but won't it still work?

**Comment by yash-rajpal** [comment]
You are right @geekgonecrazy but in latest versions we introduced a new setting and behavior that un-encrypted messages won't be allowed in encrypted rooms.

Also, currently we have no way of knowing which apps and slash-commands won't work with encrypted messages and channels, so we decided to block all apps and slashcommands.

**Comment by hugocostadev** [comment]
I also think that we need to revisit this, maybe encrypt messages sent by apps/bots if the settings that allow unencrypted messages are disabled.

**Review by hugocostadev — CHANGES_REQUESTED** [review]
Allow bots and apps if the "Unencrypted messages in encrypted rooms" is enabled

**Comment by yash-rajpal** [comment]
> I also think that we need to revisit this, maybe encrypt messages sent by apps/bots if the settings that allow unencrypted messages is enabled

I don't think that is something possible with current architecture. Also, we don't want to provide slash commands even if sending un-encrypted messages are allowed because, the room may contain encrypted messages and apps/bots won't play nice with such messages.

**Review by hugocostadev — CHANGES_REQUESTED** [review]
LGTM, but it's missing tests too 🫣

**Review by hugocostadev — CHANGES_REQUESTED** [review]
Slash commands are not working in not encrypted channels/DMs

image

File: apps/meteor/tests/e2e/e2e-encryption.spec.ts [hunk 1]
```diff
@@ -239,6 +239,44 @@ test.describe.serial('e2e-encryption', () => {
...
+
+			await expect(poHomeChannel.content.encryptedRoomHeaderIcon).toBeVisible();
+
+			await poHomeChannel.content.sendMessage('This is an encrypted message.');
+
+			await expect(poHomeChannel.content.lastUserMessageBody).toHaveText('This is an encrypted message.');
+			await expect(poHomeChannel.content.lastUserMessage.locator('.rcx-icon--name-key')).toBeVisible();
+
+			await page.locator('[name="msg"]').type('/');
+			await expect(page.locator('#popup-item-contextualbar')).toHaveClass(/disabled/)
```
[the remark this answers is not in the document]
Comment (yash-rajpal) [reply apps/meteor/tests/e2e/e2e-encryption.spec.ts]: had already fixed 🤔

**Review by hugocostadev — DISMISSED** [review]

**Review by hugocostadev — APPROVED** [review]

**Pull request merged**